python/psautohint/otfFont.py

```diff
@@ -652,24 +654,38 @@ def convertBezToT2(bezString):
     return t2Program
 
 
+def _run_tx(args):
+    try:
+        subprocess.check_call(["tx"] + args)
+    except (subprocess.CalledProcessError, OSError) as e:
+        raise FontParseError(e)
+
+
 class CFFFontData:
-    def __init__(self, path, is_otf):
+    def __init__(self, path, font_format):
         self.inputPath = path
-        self.is_otf = is_otf
+        self.font_format = font_format
 
-        if is_otf:
+        if font_format == "OTF":
             # It is an OTF font, we can process it directly.
             font = TTFont(path)
             if "CFF " not in font:
                 raise FontParseError("OTF font has no CFF table <%s>." % path)
         else:
-            # It is s CFF font, package it in an OTF font.
-            with open(path, "rb") as fp:
-                data = fp.read()
+            # Else, package it in an OTF font.
+            if font_format == "CFF":
+                with open(path, "rb") as fp:
+                    data = fp.read()
+            else:
+                fd, temp_path = tempfile.mkstemp()
+                os.close(fd)
+                _run_tx(["-cff", "+b", "-std", path, temp_path])
+                with open(temp_path, "rb") as fp:
+                    data = fp.read()
+                os.remove(temp_path)
```

Comment:
I don't think we need to bother removing a temporary file that is properly written in the system's temporary folder

Reply:
a `try: ... finally:` clause is just good practice and it's what NamedTemporaryFile would do automatically upon closing the file (if Windows didn't prevent us from using it).

Reply:
@miguelsousa I’d rather prefer to properly delete the temporary files, I have seen software that do not do so in the past and one easily ends with thousands of files in the temp directory, and this can sometimes be an issue.

Reply:
👍